core/commands/partials/terraform_cleaner.rb

```diff
@@ -14,13 +14,11 @@ def initialize(logger, aws_service)
   # @param configuration [Configuration] that we operate on
   def destroy_nodes_by_configuration(configuration)
     @ui.info('Destroying the machines using terraform')
-    if configuration.all_nodes_selected?
+    resource_type = TerraformService.resource_type(configuration.provider)
```

Comment:
In current implementation the method `TerraformService::resource_type` may throw an exception. You use this method 3 times. Please refactor this method, so it will return Result<String> instead of throwing an exception and correctly process the error state.